### PR TITLE
Add a debug panel with memory measurements.

### PR DESCRIPTION
**Pull request by SomeoneToIgnore**
The panel is opened with Ctrl + Login ( Cmd + Login for mac also) button click.

### CODE REVIEW

File: src/main/java/com/vaadin/demo/dashboard/view/MainView.java [hunk 1]
```diff
@@ -26,4 +55,60 @@ public MainView() {
...
         new DashboardNavigator(content);
     }
+
+    private void debugGarbage() {
+        System.setProperty("spectator.api.gaugePollingFrequency", "PT1S");
+        Jmx.registerStandardMXBeans(registry);
+        gc.start(new GcEventListener() {
+            @Override
+            public void onComplete(GcEvent gcEvent) {
+                System.out.println("GC happened: " + gcEvent.toString());
```
Comment (gilberto-torrezan) [inline src/main/java/com/vaadin/demo/dashboard/view/MainView.java:65]: Is this read somewhere?

File: src/main/java/com/vaadin/demo/dashboard/view/MainView.java [hunk 2]
```diff
@@ -26,4 +55,60 @@ public MainView() {
...
+
+    private void measureCurrentMemory() {
+        double totalMemoryB = 0;
+        double totalMemoryMB = 0;
+        for (Meter meter : registry) {
+            Id id = meter.id();
+            if ("jvm.memory.used".equals(id.name()) && containsTag(id.tags(), "memtype", MemoryType.HEAP.name())) {
+                for (Measurement measurement : meter.measure()) {
+                    totalMemoryB += measurement.value();
+                    totalMemoryMB += measurement.value() / 1024 / 1024;
```
Comment (gilberto-torrezan) [inline src/main/java/com/vaadin/demo/dashboard/view/MainView.java]: When you divide by 1024 * 1024, you don't have MB (megabyte), but MiB (mebibyte). To have MB you should divide by the standard 1000 * 1000.

Courtesy of the HDD manufacturers.

**Review by gilberto-torrezan — CHANGES_REQUESTED** [review]

File: src/main/java/com/vaadin/demo/dashboard/view/MainView.java (hunk 2 above)
Comment (SomeoneToIgnore) [reply src/main/java/com/vaadin/demo/dashboard/view/MainView.java]: Oh, ok, thanks.

File: src/main/java/com/vaadin/demo/dashboard/view/MainView.java (hunk 1 above)
Comment (SomeoneToIgnore) [reply src/main/java/com/vaadin/demo/dashboard/view/MainView.java:65]: Yes, in the console.
Not needed by the tests but interesting to see how many garbage and where has been removed.

**Review by gilberto-torrezan — APPROVED** [review]

**Comment by Legioth** [comment]
A much less obtrusive way would have been to deploy an additional servlet that returns a text with the current memory usage.

**Comment by SomeoneToIgnore** [comment]
We can improve it later, sure.

**Comment by SomeoneToIgnore** [comment]
Not sure about the MPR servlet – are multiple servlets supported there also?

**Comment by SomeoneToIgnore** [comment]
Also, current method is easy enough to use manually by just opening the website.

Not sure about obtrusiveness also – the panel is not shown for simple login.

**Pull request merged**